spec/hcloud/server_spec.rb

```diff
@@ -63,5 +67,34 @@
     expect(server.created).to be_a Time
     expect(action.status).to eq('running')
     expect(pass).to eq('moo')
+    expect(server.volumes).to be_a Array
+    expect(server.volumes).to be_empty
+  end
+
+  it 'get server with volumes' do
+    related_action = new_action.merge(action_status(:running))
+    servers[1][:volumes] = volumes.map { |volume| volume[:id] }
+    stub_collection("servers/#{servers[1][:id]}/actions", [related_action], resource_name: :actions)
+    stub('servers') do |_request, _page_info|
+      {
+        body: { server: servers[1], action: related_action, root_password: :moo },
+        code: 201
+      }
+    end
+    action, server, pass = nil
+    expect do
+      action, server, pass = client.servers.create(name: 'moo', server_type: 'cx11', image: 1)
+    end.not_to(raise_error)
+    expect(server.volumes).to be_a Array
+    expect(server.volumes).to_not be_empty
+
+    stub("volumes/#{volumes[0]['id']}") do |_request, _page_info|
+      {
+        body: { volume: volumes[0] },
+        code: 200
+      }
+    end
+    expect(server.volumes.first).to eq(volumes[0])
+    expect(server.volumes).to eq(volumes)
```

Comment:
While implementing my server-volume test, I get a stack overflow due endless recursion. Do I use the stub in the wrong way? @tonobo

Reply:
not fully sure about it, but the future seems to have trouble?